lib/AST/UnqualifiedLookup.cpp

```diff
@@ -400,6 +400,10 @@ bool implicitSelfReferenceIsUnwrapped(const ValueDecl *selfDecl,
   // Find the condition that defined the self decl,
   // and check that both its LHS and RHS are 'self'
   for (auto cond : conditionalStmt->getCond()) {
+    if (cond.getKind() != StmtConditionElement::CK_PatternBinding) {
+      continue;
+    }
+    
     if (auto pattern = cond.getPattern()) {
```

Comment:
There is `getPatternOrNull` so you don’t have to check kind.

Reply:
Any clue why `cond.getInitializerOrNull()` doesn't exist? Alternatively we could just add it, and check `getPatternOrNull()` and `getInitializerOrNull()` instead of checking the kind.

Reply:
Let's add `getInitializerOrNull` and check it there yeah.

Reply:
Could we actually package this logic into a function and call from both places? I noticed that one handles `LoadExpr` for initializer and this one doesn't so it would be good to unify the logic.

Reply:
Sure, that seems like a good idea @xedin. Any suggestion on where to put it the shared helper? We could define this as a public method on `ValueDecl` like `ValueDecl::allowsImplicitSelfInWeakSelfClosure()` or a check on `LabeledConditionalStmt` like `LabeledConditionalStmt::rebindsSelf`. Of those I'd probably lean towards `LabeledConditionalStmt::rebindsSelf`. Open to other suggestions as well.

Reply:
Let's add `bool StmtConditionElement::rebindsSelf` so that `LabeledConditionalStmt::rebindsSelf` could be simply something like `return llvm::any_of(getCond(), [](const auto &cond) { return cond.rebindsSelf() });`

Reply:
@xedin, consolidated this logic into a shared `LabeledConditionalStmt::rebindsSelf` helper. Feels like a nice simplification.